luxonis_train/core/core.py

```diff
@@ -429,15 +433,20 @@ def infer(
         @param save_dir: Directory where to save the visualizations. If
             not specified, visualizations will be rendered on the
             screen.
+        @type img_path: str | None
+        @param img_path: Path to the image file or directory for inference.
+            If None, defaults to using dataset images.
         """
         self.lightning_module.eval()
 
-        for inputs, labels in self.pytorch_loaders[view]:
-            images = get_unnormalized_images(self.cfg, inputs)
-            outputs = self.lightning_module.forward(
-                inputs, labels, images=images, compute_visualizations=True
-            )
-            render_visualizations(outputs.visualizations, save_dir)
+        if img_path:
+            img_path_obj = Path(img_path)
+            if img_path_obj.is_file():
+                process_single_image(self, img_path_obj, view, save_dir)
```

Comment:
Can these two be merged: single image can be packaged in a list of len=1 and represetn a "directory of images". And then dummy labels can be generated only once for whole directory and process_single_image would take them as parameter. This would optimize the label generation a bit as currently we re-generate them for every image in the directory. Thoughts?